runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -308,28 +309,32 @@ getJ9UTF8SignatureFromMethodType(J9VMThread *currentThread, j9object_t typeObjec
 		if (NULL == rSignature) {
 			goto done;
 		}
-	}
 
-	signatureLength += strlen(rSignature);
+		signatureLength += strlen(rSignature);
+	}
 
 	methodDescriptor = (J9UTF8*)j9mem_allocate_memory(signatureLength + sizeof(J9UTF8), OMRMEM_CATEGORY_VM);
 	if (NULL == methodDescriptor) {
 		goto done;
 	}
 	J9UTF8_SET_LENGTH(methodDescriptor, (U_16)signatureLength);
 	cursor = (char*)J9UTF8_DATA(methodDescriptor);
+	space = signatureLength;
 	*cursor++ = '(';
+	space -= 1;
 
 	/* Copy class signatures to descriptor string */
 	for (U_32 i = 0; i < numArgs; i++) {
 		UDATA len = strlen(signatures[i]);
-		strncpy(cursor, signatures[i], len);
+		strncpy(cursor, signatures[i], space);
```

Comment:
It's not very efficient, as it's padding with nulls to `space` every time through the loop.

Reply:
It's always seemed unnatural to me that strncpy has that behavior, but you're right, I'll fix that.